src/FshCompletionProvider.ts

```diff
@@ -131,6 +138,199 @@ export class FshCompletionProvider implements CompletionItemProvider {
+      if (checkedEntities.includes(entityToCheck)) {
+        // circular reference escape hatch
+        return null;
+      }
+      checkedEntities.push(entityToCheck);
+      const entityInfo = this.definitionProvider.nameInformation.get(entityToCheck);
+      // take the first potentially useful thing from entityInfo
+      const infoToUse = entityInfo.find(info =>
+        ['Profile', 'Extension', 'Resource', 'Logical', 'Instance'].includes(info.type)
+      );
+      if (infoToUse == null) {
+        // the entity may be an alias that we are able to resolve
+        const aliasToUse = entityInfo.find(info => info.type === 'Alias');
+        if (aliasToUse) {
+          entityToCheck = aliasToUse.aliasValue;
+        } else {
+          // otherwise, we don't know
+          return null;
+        }
+      } else {
+        entityToCheck = infoToUse.parent ?? infoToUse.instanceOf;
+        // special handling for default parents:
+        // an Extension's default parent is Extension
+        // a Resource's default parent is DomainResource
+        // a Logical's default parent is Base
+        if (entityToCheck == null) {
+          if (infoToUse.type === 'Extension') {
+            entityToCheck = 'Extension';
+          } else if (infoToUse.type === 'Resource') {
+            entityToCheck = 'DomainResource';
+          } else if (infoToUse.type === 'Logical') {
+            entityToCheck = 'Base';
+          }
+        }
+      }
+    }
+    // is the entity present in the dependencies' entities?
+    // we may have to search twice, if we find a Profile
+    let originalType: string;
+    for (const [, entities] of this.fhirEntities) {
+      if (entities.profiles.has(entityToCheck)) {
+        originalType = entities.profiles.get(entityToCheck).baseDefinition;
+        break;
+      }
+      if (entities.extensions.has(entityToCheck)) {
+        return entities.extensions.get(entityToCheck).elements;
+      }
+      if (entities.resources.has(entityToCheck)) {
+        return entities.resources.get(entityToCheck).elements;
+      }
+      if (entities.logicals.has(entityToCheck)) {
+        return entities.logicals.get(entityToCheck).elements;
+      }
+    }
+    if (originalType != null) {
+      // search again, but don't include Profiles
+      for (const [, entities] of this.fhirEntities) {
+        if (entities.extensions.has(originalType)) {
+          return entities.extensions.get(originalType).elements;
+        }
+        if (entities.resources.has(originalType)) {
+          return entities.resources.get(originalType).elements;
+        }
+        if (entities.logicals.has(originalType)) {
+          return entities.logicals.get(originalType).elements;
+        }
+      }
+    }
+    return null;
+  }
+
+  public getPathItems(existingPath: string[], baseElements: ElementInfo[]): CompletionItem[] {
+    let targetElement: ElementInfo;
+    let targetLevel = baseElements;
```

Comment:
I'm not sure I understand what `targetElement` and `targetLevel` represent, particularly `targetLevel`.  It's probably just me, but is there a different name that might work better?